src/main/java/seedu/address/model/task/deadline/Deadline.java

```diff
@@ -107,10 +107,11 @@ private boolean isSameDeadline(Deadline otherDeadline) {
         if (otherDeadline == this) {
             return true;
         }
-
         return otherDeadline != null
                 && otherDeadline.getTitle().equals(getTitle())
-                && otherDeadline.getDeadlineDateTime().equals(getDeadlineDateTime());
+                && otherDeadline.getDeadlineDateTime().equals(getDeadlineDateTime())
```

Comment:
This is repeated code, just renamed in both deadline and events, since both are a sub class of task, implement these methods in task and you can override later on